auth/authentication.go

```diff
@@ -5,41 +5,62 @@ import (
 	"net/http"
 
 	"github.com/ONSdigital/go-ns/log"
+	"github.com/ONSdigital/dp-search-api/models"
 )
 
 // Authenticator structure which holds the secret key for validating clients. This will be replaced in the future, after the `thin-slices` has been delivered
 type Authenticator struct {
-	SecretKey  string
-	HeaderName string
+	SecretKey  				string
+	HeaderName 				string
+	HasPrivateEndpoints     bool
 }
 
 // Check wraps a HTTP handle. If authentication fails an error code is returned else the HTTP handler is called
 func (a *Authenticator) Check(handle func(http.ResponseWriter, *http.Request)) http.HandlerFunc {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		key := r.Header.Get(a.HeaderName)
-		if key == "" {
-			http.Error(w, "No authentication header provided", http.StatusUnauthorized)
-			log.Error(errors.New("client missing token"), log.Data{"header": a.HeaderName})
-			return
-		}
-		if key != a.SecretKey {
-			http.Error(w, "Unauthorised access to API", http.StatusUnauthorized)
-			log.Error(errors.New("unauthorised access to API"), log.Data{"header": a.HeaderName})
+
+		// If we're not in in the publishing subnet. Nothing has auth
```

Comment:
This comment maybe true but this will never get run due to if statement in api.go, see above comment for why this package does not need to be changed